### ISSUE TITLE
Issues with Google Play Music Desktop Player on mediaplayer

### ISSUE THREAD

**Issue by WolfangAukang** [vivien/i3blocks-contrib#91]
Google Play Music Desktop Player works fine on playerctl, so I decided to test it as an instance name on i3blocks (it works on playerctl as google-play-music-desktop-player, according to the dbus mpris service).

I understand that this can be solved without specifying instance, but this behavior is not the correct.

### Expected behavior
Shows the artist name and the song played

### Actual behavior
(null) - (null)

The settings I have on i3blocks config file on mediaplayer are the following:
```
[mediaplayer]
instance=google-play-music-desktop-player
interval=5
signal=10
```

### Relevant commands

- `playerctl --player='google-play-music-desktop-player' metadata artist`
*Shows some warnings but at the end shows the artist*

- `playerctl metadata artist`
*Shows some warnings but at the end shows the artist*

- `./mediaplayer`
*Shows the artist correctly, because instance is not specified, so it works as the case before*

- Executing with the settings indicated on config file
`(null) - (null)`

### Pictures
Without instance
![capture](https://cloud.githubusercontent.com/assets/8378365/25913789/48202198-3579-11e7-93e7-974cf1dfb7b9.PNG)

With instance google-play-music-desktop-player
![capture](https://cloud.githubusercontent.com/assets/8378365/25913856/816d6aa0-3579-11e7-9f86-b1615bbf5539.PNG)

**Comment by kb100** [comment]
Hi @WolfangAukang, at first glance, the blocklet should work with the google-play-music-desktop-player exactly as you have specified. Could you copy the exact warnings that it shows when you say "Shows some warnings but..."?

Could you also clarify where the issue is occurring? You say `./mediaplayer` shows the artist correctly. If you do not specify a block instance, then the script defaults to using `mpd` backend, so I don't think it shouid work in that case.  Also what is the behavior if you run `BLOCK_INSTANCE=google-play-music-desktop-player ./mediaplayer`?

Could you also check the exit statuses of the previous commands you ran? E.g.

    playerctl --player='google-play-music-desktop-player' metadata artist
    echo $?

**Comment by WolfangAukang** [comment]
Thanks for the response @kb100 

1. The warning messages are the following:
```
(playerctl:13598): GLib-GIO-WARNING **: Received property Rate with type y does not match expected type d in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property Volume with type y does not match expected type d in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property Position with type u does not match expected type x in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property MinimumRate with type y does not match expected type d in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property MaximumRate with type y does not match expected type d in the expected interface
artist-name
```

2. 
```
(playerctl:13598): GLib-GIO-WARNING **: Received property Rate with type y does not match expected type d in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property Volume with type y does not match expected type d in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property Position with type u does not match expected type x in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property MinimumRate with type y does not match expected type d in the expected interface

(playerctl:13598): GLib-GIO-WARNING **: Received property MaximumRate with type y does not match expected type d in the expected interface
artist-name - song-name
```

3. I receive a 0

**Comment by kb100** [comment]
Hmm, all the script does is call the command you already mentioned that works, so this is a little wonky.  Can you tell me what happens when you run the script as `BLOCK_INSTANCE=google-play-music-desktop-player ./mediaplayer`? Then can you try adding print statements to the script in the playerctl function and trying to figure out where things differ from what they are supposed to be? I don't have the google play music desktop player otherwise I would do this myself. The more info you can provide the better.

**Comment by WolfangAukang** [comment]
Alright, I think I found what's happening. I printed the `player_arg` variable, the `player $player_arg metadata artist` command and the metadata output.

The i3blocks config for the media player is this one:
```
[mediaplayer]
instance=google-play-music-desktop-player
interval=5
signal=10
```

The block instance text is being cut, maybe for being too large. Instead of sending 'google-play-music-desktop-player', it cuts the r from the ending, sending 'google-play-music-desktop-playe'.

For example, the print I have at the i3blocks is the following:
![2017-05-13 19 28 39](https://cloud.githubusercontent.com/assets/8378365/26030666/36e2ac12-3816-11e7-9ff4-a93a29ca2ad4.png)

With Spotify, this doesn't happen, as shown here:
![2017-05-13 19 34 15](https://cloud.githubusercontent.com/assets/8378365/26030668/3b45571e-3816-11e7-9013-2deef2804e89.png)

**Comment by kb100** [comment]
@WolfangAukang are you using the most recent version of i3blocks? A year ago the instance variable was limited to 32 characters, which is exactly the length yours is getting cut off at. This was updated to 256 in https://github.com/vivien/i3blocks/commit/6a345440853d1cd16cef7096ffc636bed161fd91

**Comment by WolfangAukang** [comment]
I am using the one on Arch Linux repositories (not the AUR one, if there's any)

**Comment by kb100** [comment]
Indeed, that Arch official one is using 1.4-4 apparently, wheras the fix was added to i3blocks in 1.4-17. The one in AUR is 1.4-21 so using that version should fix your problem.
